### DISCUSSION TITLE
Running with uwsgi

### DISCUSSION THREAD

**Discussion by javawolfpack** [INGInious/INGInious#945] [Q&A]
So added an issue as do feel like it's missing documentation so falls under as an issue: https://github.com/UCL-INGI/INGInious/issues/944

How do you run this as uwsgi? Everything I've tried has failed so far. It almost feels like I need to write my own wsgi.py file but it would be nice to know that's the case or since it's already been solved if someone could share. Thanks. 

Seems to be the case with both the inginious-webapp and inginious-webdav, and I guess the inginious-webterm if I get there.

**Comment by javawolfpack** [comment]
```ini
[uwsgi]
#environment variables
env  = INGINIOUS_WEBAPP_HOST=0.0.0.0
env  = INGINIOUS_WEBAPP_PORT=8080
env  = INGINIOUS_WEBAPP_CONFIG=/var/www/inginious/configuration.yaml
#socket details
socket = /tmp/inginious-webapp.sock
processes = 2
threads = 2
#module and callable
module = inginious
callable = inginious-webapp
#extra settings
http-timeout = 86400
http-timeout-as = 86400
vacuum = true
die-on-term = true
master = true
chdir = /var/www/inginious/
#user and group to run socket ownership for
uid = www-data
gid = www-data
# set max connections to 1024 in uWSGI
listen = 1024 
```
Where I'm at currently, can't seem to figure out what to put as the callable, or if the module/callable are even on the right track. Running this as the following:

```bash
$ uwsgi --ini inginious_webapp.ini
```

**Comment by anthonygego** [comment]
I suggest that we continue this discussion in #944. FYI, `inginious-webterm` is a very small Node.js app because there was a ready-to-use library available at the time of writing it. We did not investigate the ability to do it using Python with time.

Node.js apps are designed to be run standalone but you can also run them behind a webserver using proxy modules to share the same port as another app for instance.
